### DISCUSSION TITLE
Upgrade more than one package but not all of them

### DISCUSSION THREAD

**Discussion by rafrafek** [pypa/pipx#1317] [Q&A] [answered]
Hello,

How to run `pipx upgrade` on more than one package but not on all of them? Or how to upgrade all packages but not one? I need to upgrade everything but leave an older version of Poetry not updated.

```shell
➜  ~ pipx --version
1.5.0
➜  ~ pipx upgrade aws-sso-util awsume black pylint pyupgrade
usage: pipx [-h] [--quiet] [--verbose] [--global] [--version]
            {install,uninject,inject,upgrade,upgrade-all,uninstall,uninstall-all,reinstall,reinstall-all,list,interpreter,run,runpip,ensurepath,environment,completions} ...
pipx: error: unrecognized arguments: awsume black pylint pyupgrade
➜  ~ 
```

There is https://github.com/pypa/pipx/pull/1102 but it's only about installing, not upgrading.

**Comment by dukecat0** [comment, marked as answer]
You can run `pipx upgrade-all --skip <packages you want to skip>` for now. However, I think it's a good idea to allow `pipx upgrade` to accept multiple packages at the same time, just like the `install` command.

**Reply by rafrafek** [reply]
Thank you! The `--skip` option works great in my case!

**Comment by uranusjr** [comment]
With the `--skip` thing mentioned, personally I’m not sure why `upgrade` can’t take multiple apps myself either… It seems to make sense.
